src/util/mpir_pmi.c

```diff
@@ -989,6 +1004,22 @@ static int build_nodemap_roundrobin(int num_cliques, int *nodemap, int sz, int *
     return MPI_SUCCESS;
 }
 
+/* assign processes to num_cliques nodes by uniform block */
+static int build_nodemap_byblock(int num_cliques, int *nodemap, int sz, int *p_max_node_id)
+{
+    int block_size = sz / num_cliques;
+    int remainder = sz % num_cliques;
+    for (int i = 0; i < sz; ++i) {
+        if (i < remainder) {
+            nodemap[i] = i / (block_size + 1);
+        } else {
+            nodemap[i] = (i - remainder) / block_size;
+        }
```

Comment:
What if `i=0`, `remainder=2`, and `block_size=2`? Negative node id?

Reply:
that is `i < remainder` branch, `node_id` is `0` in that case.

Reply:
I think it is fixed now.

Reply:
Yeah, this took me a minute but I get it now. LGTM.